src/lib/hooks/gun/useGunProfile.js

```diff
@@ -0,0 +1,55 @@
+import { useEffect, useRef, useState } from 'react'
+import { get, result } from 'lodash'
+import userStorage from '../../gundb/UserStorage'
+import isEmail from '../../validators/isEmail'
+import isMobilePhone from '../../validators/isMobilePhone'
+import logger from '../../logger/pino-logger'
+
+const log = logger.child({ from: 'useGunProfile' })
+
+const useGunProfile = (identifier, fields = ['fullName', 'smallAvatar']) => {
+  const attr = isMobilePhone(identifier) ? 'mobile' : isEmail(identifier) ? 'email' : 'walletAddress'
+  const value = userStorage.constructor.cleanHashedFieldForIndex(attr, identifier)
+  const gunSubscriptions = useRef({})
+  const [profile, setProfile] = useState({})
+
+  const onProfileField = field => (data, nodeID, message, event) => {
+    log.debug('got field:', { data, field })
+    gunSubscriptions.current[field] = { event, value: data }
+    const updated = {}
+    fields.forEach(f => (updated[f] = get(gunSubscriptions, `current.${f}.value`)))
+    setProfile(updated)
+  }
+
+  const unsubscribe = () => {
+    fields.forEach(field => {
+      log.debug('unsubscribe field:', { field })
+
+      result('gunSubscriptions.current', `${field}.event.off`)
+      delete gunSubscriptions.current[field]
+    })
+  }
+
+  useEffect(() => {
+    if (!identifier || !fields) {
```

Comment:
hooks count should be always the same.
if you need to return cancel hook - return it **always**
just add the same check to it and don't do anything if check failed

Reply:
why? i've never seen it cause any issue.
if there's no cleanup needed why return the cleanup function?
where have you seen this mentioned?